### ISSUE TITLE
There seems to be no code for global_filter in the data processing code

### ISSUE THREAD

**Issue by Jack-Yang-S** [ymoslem/Adaptive-MT-LLM-Fine-tuning#1]
Dear author, it seems that the global_filter existing in the data processing code can not find the source code, may I ask whether the code needs to be modified

**Comment by ymoslem** [comment]
Hello! I have added the file here ([global_filter.py](https://github.com/ymoslem/Adaptive-MT-LLM-Fine-tuning/blob/main/global_filter.py)). It mainly removes duplicates between the combined dataset, and applies a length limit of 70 words.

The reasons I filter individual datasets first and then apply a global filter are: 1) it is more efficient to filter smaller datasets, and 2) in the global filter, I usually have harsher settings (e.g. length limit), so I can always revert to the individually filtered datasets if needed. So, it is up to you if you want to apply a global filter directly rather than filtering each dataset individually first, or do it the way I did.

All the best,
Yasmin
